rs/nervous_system/clients/src/ledger_client.rs

```diff
@@ -114,4 +115,12 @@ impl ICRC1Ledger for LedgerCanister {
         let principal_id = PrincipalId::from(self.client.ledger_canister_id);
         CanisterId::unchecked_from_principal(principal_id)
     }
+
+    // Shah-TODO: if this function should be implemented?
+    async fn icrc3_get_blocks(
```

Comment:
I think this is the one that we need to implement, since we're mostly interested in the SNS ledger (not the ICP ledger). 

Please note that `ICRC1Ledger` refers to any canister deployed with the Wasm from this target: `//rs/ledger_suite/icrc1/ledger:ledger_canister`

In case we don't want to implement this function for one of the ledgers, I would suggest that we still just return an error rather than panick (which is what the `unimplemented!()` macro would do). Panicking from the canister can be dangerous because the state is rolled back only up to the last async call, so another programmer might misuse this function and introduce a bug if it panics.